src/lwc/rd2EntryForm/rd2EntryForm.js

```diff
@@ -206,13 +185,18 @@ export default class rd2EntryForm extends LightningElement {
         event.preventDefault();
         const fields = event.detail.fields;
 
-        const scheduleSection = this.template.querySelector('c-rd2-entry-form-schedule-section');
+        const scheduleSection = this.scheduleComponent;
         const scheduleFields = (scheduleSection === null || scheduleSection === undefined)
             ? {}
             : scheduleSection.returnValues();
 
+        const donorSection = this.donorComponent;
+        const donorFields = (donorSection === null || donorSection === undefined)
+            ? {}
+            : donorSection.returnValues();
+
```

Comment:
You can add a let isValid = donorSection.validateFields(); in here to check the required field missing validation, and add the following function in the rd2EntryFormDonorSection.js 
` @api
    validateFields() {
    let isValid = true;
    this.template.querySelectorAll('lightning-input-field')
            .forEach(field => {
                isValid = field.reportValidity()
            });
        return isValid;
    }`

Then only submit the form if isValid = true

Reply:
Thanks @andrewyu-salesforce. Changes made.